app/src/webkit/java/org/mozilla/focus/webkit/WebkitView.java

```diff
@@ -396,7 +399,28 @@ public void onDownloadStart(String url, String userAgent, String contentDisposit
                 if (downloadCallback != null) {
                     final String name = URLUtil.guessFileName(url, contentDisposition, mimetype);
                     final Download download = new Download(url, name, userAgent, contentDisposition, mimetype, contentLength, false);
-                    downloadCallback.onDownloadStart(download);
+                    AlertDialog.Builder localBuilder = new AlertDialog.Builder(getContext());
```

Comment:
Please implement the dialog in DownloadCallback's implementation like Browserfragment.
WebkitView shouldn't have UI logic.